### PR TITLE
Add option to filter by `measurand`

### PR DESCRIPTION
**Pull request by danielhuppmann**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- ~Name of contributors Added to AUTHORS.rst~
- [x] Description in RELEASE_NOTES.md Added

# Description of PR

This PR adds a feature to support filtering by a "measurand", i.e., a tuple (or iterable of tuples) of variable and unit

See the related discussion at https://github.com/IAMconsortium/nomenclature/pull/360

### CODE REVIEW

File: tests/test_core.py [hunk 1]
```diff
@@ -447,6 +447,19 @@ def test_filter_variable_and_depth(test_df, filter_args):
...
 
+def test_filter_measurand_list(test_df):
+    data = test_df.data
+    data.loc[3, "variable"] = "foo"
+    data.loc[4, "unit"] = "bar"
+    df = IamDataFrame(data)
+
+    obs = df.filter(measurand=(("foo", "EJ/yr"), ("Primary Energy", "bar")))
+
+    assert obs.variable == ["foo", "Primary Energy"]
```
Comment (phackstock) [inline tests/test_core.py]: Comparisons where unintentionally the order matters are always prone to breaking, I'd change it to:
```suggestion
    assert set(obs.variable) == {"foo", "Primary Energy"}
```

File: tests/test_core.py [hunk 2]
```diff
@@ -447,6 +447,19 @@ def test_filter_variable_and_depth(test_df, filter_args):
...
+    assert obs.unit == ["EJ/yr", "bar"]
```
Comment (phackstock) [inline tests/test_core.py:459]: Same as above, using a set means the test won't break if the order of units changes

```suggestion
    assert set(obs.unit) == {"EJ/yr", "bar"}
```

**Review by phackstock — APPROVED** [review]
One small comment, after that good to merge.

File: tests/test_core.py [hunk 3]
```diff
@@ -431,14 +431,35 @@ def test_filter_empty_df():
...
 
+def test_filter_measurand_list(test_df):
+    data = test_df.data
+    data.loc[4, "variable"] = "foo"
+    data.loc[5, "unit"] = "bar"
+    df = IamDataFrame(data)
+
+    obs = df.filter(measurand=(("foo", "EJ/yr"), ("Primary Energy", "bar")))
+
+    assert obs.variable == ["Primary Energy", "foo"]
```
Comment (danielhuppmann) [inline tests/test_core.py:458]:
```suggestion
    assert set(obs.variable) == {"Primary Energy", "foo"}
```

**Pull request merged**